src/app/shared/modules/create-product-modal/create-product-modal.component.ts

```diff
@@ -39,11 +40,38 @@ export class CreateProductModalComponent implements OnDestroy {
           Validators.maxLength(500),
           Validators.pattern(TEXT_REGEX_PATTERN)
         ]
-      ]
+      ],
+      counts: this.fb.array([
+        this.fb.group({
+          location: [''],
+          quantityAvailable: ['', Validators.pattern(NUMERIC_REGEX_PATTERN)],
+          price: ['', Validators.pattern(NUMERIC_REGEX_PATTERN)]
+        })
+      ])
     },
     { updateOn: 'blur' }
   );
 
+  counts(): FormArray {
+    return this.form.get('counts') as FormArray;
```

Comment:
In general, it is not a good style to write single-line methods. Please use directly *this.form.get('counts')* and do not use ***as*** casting - it is forbidden and considered as bad practice.